cli/src/deno.rs

```diff
@@ -81,9 +84,29 @@ pub async fn run(extension_state: ExtensionState, entry_point: &str) -> Result<(
 }
 
 /// See https://github.com/denoland/deno/blob/main/core/examples/ts_module_loader.rs.
-struct TypescriptModuleLoader;
+struct PhylumExtensionsModuleLoader;
+
+impl PhylumExtensionsModuleLoader {
+    async fn load_from_filesystem(path: &Path) -> Result<String> {
```

Comment:
From testing, it appears that it gets normalized into an absolute path by `deno_core`. Not sure how strong of a guarantee we have on that front, but otoh I think it would be pretty inconvenient for an implementor of the trait to have to deal with relative URLs anyway, so I believe we can rely on this.

Added note: not only local paths are passed in as absolute, but also remote ones.

Reply:
I feel like it's probably safer to check/redo normalization ourselves, since this is somewhat of a security check and we wouldn't want people to be able to bypass it with symlinks?

Reply:
> Wrt renormalization: are we confident about the fact that a relative path is relative to the current working directory?

I'm not sure what you mean? Are you saying that we don't necessarily know that the relative path was intended to be relative to our applications current binary, instead of relative to the loaded module?

Honestly I'd say it doesn't even matter that much, if deno normalizes things already, just doing `!relative && !symlink` would be enough for me. My goal is just that we're safe from the worst-case scenario, I don't really care if we just explode in those situations.

> Anyway, I found a guarantee that the path is absolute here

That's no guarantee, it's inside of `deno_core` and thus might change at any time.

Reply:
> Honestly I'd say it doesn't even matter that much, if deno normalizes things already, just doing !relative && !symlink would be enough for me. My goal is just that we're safe from the worst-case scenario, I don't really care if we just explode in those situations.

Agreed 100%

> That's no guarantee, it's inside of deno_core and thus might change at any time.

The type is an alias to `url::Url` which is from a separate crate. Any change to that behaviour would most definitely break our build. If we want to be extra sure, we could just add tests for it.

Reply:
> The type is an alias to url::Url which is from a separate crate. Any change to that behaviour would most definitely break our build. If we want to be extra sure, we could just add tests for it.

They could easily change that to use `Url::parse` and `Path::as_str` or whatever. Just advocating to better be safe than sorry.

Reply:
Being better safe than sorry is something I fully agree with, although I'm not sure how we could hedge our bets in this specific case -- can't find a way of determining if a [`Url`](https://docs.rs/url/latest/url/struct.Url.html) object is relative or absolute, and the documentation in general seems to suggest there is no state where relative paths are valid. We do use methods from `Url` so if they changed the type we'd get a build error.

Reply:
We get a path though, not a url, right? All we have to do is check that path?

Reply:
No, the other way around -- we get a `Path` from the `Url` [here](https://github.com/phylum-dev/cli/blob/andrea/module-loader/cli/src/deno.rs#L153) which is stated to be [absolute](https://docs.rs/url/latest/url/struct.Url.html#method.to_file_path).